### PR TITLE
Add theme keys for the picker header area

### PR DESCRIPTION
**Pull request by ofzo**
full fill column's background for picker, it can be defined use 
```toml
"ui.picker.header" = { fg= "white", bg = "red" }
```
image

### CODE REVIEW

File: helix-term/src/ui/info.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ impl Component for Info {
             width,
             height,
         ));
-        surface.clear_with(area, popup_style);
+        surface.clear_with(area, text_style);
```
Comment (the-mikedavis) [inline helix-term/src/ui/info.rs:24]: It looks like this is an unrelated change? I believe `popup_style` is the correct style to clear with here anyways

File: helix-term/src/ui/picker.rs [hunk 2]
```diff
@@ -694,6 +694,14 @@ impl<T: 'static + Send + Sync, D: 'static + Send + Sync> Picker<T, D> {
...
             }
         }
+        let header_style = cx.editor.theme.get("ui.picker.header");
+        if self.columns.len() > 1 {
+            for x in inner.left()..inner.right() {
+                if let Some(cell) = surface.get_mut(x, inner.y + 2) {
+                    cell.set_symbol(" ").set_style(header_style);
+                }
+            }
+        }
```
Comment (the-mikedavis) [inline helix-term/src/ui/picker.rs:704]: Instead of manually painting the cells you can set a style for the row with `Row::style` - you can set that in the block marked with `// -- Header` below - and the table widget will take care of applying the style to the header area.

```rust
let header = Row::new(self.columns.iter().map(...)).style(header_style);
table = table.header(header);
```

**Review by the-mikedavis — COMMENTED** [review]
Using the `ui.picker.header` key will change the meaning for it that I had in mind when introducing it. The current behavior is that the `bg` or modifiers (underlined for example) will only affect the column names. For example in my theme I have `"ui.picker.header" = { underline.style = "line" }`:

before

But if we apply the style to the header area:

after

I could see both ways being useful so I think we should change the keys around so that both are possible. Let's use `ui.picker.header` for the header (`Row`)'s style and introduce `ui.picker.header.column` that replaces the current use of `ui.picker.header`. Then `ui.picker.header.active` can move to `ui.picker.header.column.active`

We usually try to avoid breaking changes but these theme keys were merged very recently and no themes are using it in the `runtime/themes` dir yet so I think it's fine in this case.

File: helix-term/src/ui/info.rs (hunk 1 above)
Comment (ofzo) [reply helix-term/src/ui/info.rs]: oh, yes, this is an unrelated change, it's my mistake. I will revert it.

**Comment by ofzo** [comment]
@the-mikedavis I have resolve it.

**Review by the-mikedavis — COMMENTED** [review]
Looks good codewise. We just need to update the docs about these keys too: https://github.com/helix-editor/helix/blob/6eae8461978a5fd84387d7bb7b4021e2fa48fa31/book/src/themes.md?plain=1#L300-L301

File: book/src/themes.md [hunk 3]
```diff
@@ -293,12 +293,13 @@ These scopes are used for theming the editor interface:
...
-| `ui.bufferline.active`            | Style for the active buffer in buffer line                                                        |
+| `ui.bufferline.active`            | Style for the active buffer in buffer line                                                     |
 | `ui.bufferline.background`        | Style for bufferline background                                                                |
 | `ui.popup`                        | Documentation popups (e.g. Space + k)                                                          |
 | `ui.popup.info`                   | Prompt for multiple key options                                                                |
-| `ui.picker.header`                | Column names in pickers with multiple columns                                                  |
-| `ui.picker.header.active`         | The column name in pickers with multiple columns where the cursor is entering into.            |
+| `ui.picker.header`                | Header row names in pickers with multiple columns                                              |
```
Comment (the-mikedavis) [inline book/src/themes.md:300]:
```suggestion
| `ui.picker.header`                | Header row area in pickers with multiple columns                                              |
```

since this covers the whole header area rather than just the column names

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]
